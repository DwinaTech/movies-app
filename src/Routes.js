import React from "react";
import SideBar from "./SideBar";
import Moves from "./components/Moves";
import { Container, Row, Col } from "react-bootstrap";
import { Route, BrowserRouter as Router } from "react-router-dom";
import "./routes.css";

function Routes() {
  return (
    <Router>
      <Container fluid>
        <Row>
          <Col className="side-bar" xs={3}>
            <SideBar />
          </Col>
          <Col className="main-contents" xs={9}>
            <Route exact path="/" render={() => <h1>Home component</h1>} />
            <Route path="/moves" component={Moves} />
            <Route
              path="/favorite"
              render={() => <h1>Favorite component</h1>}
            />
          </Col>
        </Row>
      </Container>
    </Router>
  );
}

export default Routes;
