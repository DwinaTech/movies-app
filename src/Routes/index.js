import React from "react";
import SideBar from "../components/SideBar";
import Home from "../components/Home";
import movies from "../components/Movies";
import data from "../data/movies.json";
import AddMovie from "../components/AddMovie";
import Favourites from "../components/Favourites";
import { Container, Row, Col } from "react-bootstrap";
import { storeMoviesData, getMoviesData } from "../helpers";
import { Route, BrowserRouter as Router } from "react-router-dom";
import "./routes.css";

function Routes() {
  const filmsData = Object.keys(getMoviesData() || {});
  if (!filmsData || !filmsData.length) {
    storeMoviesData(data);
  }

  return (
    <Router>
      <Container>
        <Row>
          <Col className="side-bar-container" xs={3} md={2}>
            <SideBar />
          </Col>
          <Col className="main-contents" xs={9} md={10}>
            <Route exact path="/" component={Home} />
            <Route path="/movies" component={movies} />
            <Route path="/favourite" component={Favourites} />
            <Route path="/add-movie" component={AddMovie} />
          </Col>
        </Row>
      </Container>
    </Router>
  );
}

export default Routes;
