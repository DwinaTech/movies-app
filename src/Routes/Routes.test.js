import React from 'react';
import TestRenderer from 'react-test-renderer';
import Routes from './index';

it('renders without crashing', () => {
  const wrapper = TestRenderer.create(<Routes />);
  expect(wrapper).toMatchSnapshot();
});
