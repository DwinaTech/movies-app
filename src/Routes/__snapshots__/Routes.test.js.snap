// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="side-bar-container col-md-2 col-3"
    >
      <div
        className="flex-column side-bar nav nav-pills"
        onKeyDown={[Function]}
      >
        <div
          className="nav-item"
        >
          <a
            aria-disabled={true}
            className="nav-link disabled"
            data-rb-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={-1}
          >
            movies:
          </a>
        </div>
        <div
          className="nav-item"
        >
          <a
            className="nav-link active"
            data-rb-event-key="/"
            disabled={false}
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </div>
        <div
          className="nav-item"
        >
          <a
            className="nav-link"
            data-rb-event-key="/movies"
            disabled={false}
            href="/movies"
            onClick={[Function]}
          >
            movies
          </a>
        </div>
        <div
          className="nav-item"
        >
          <a
            className="nav-link"
            data-rb-event-key="/favourite"
            disabled={false}
            href="/favourite"
            onClick={[Function]}
          >
            favourite
          </a>
        </div>
        <div
          className="nav-item"
        >
          <a
            className="nav-link"
            data-rb-event-key="/add-movie"
            disabled={false}
            href="/add-movie"
            onClick={[Function]}
          >
            Add Film
          </a>
        </div>
      </div>
    </div>
    <div
      className="main-contents col-md-10 col-9"
    >
      <div
        className="movies row"
      >
        <div
          className="col-md-8 col-sm-12 col-12"
        >
          <form
            className=""
          >
            <div
              className="form-group row"
            >
              <label
                className="form-label col-form-label col-sm-1"
                htmlFor="formHorizontalEmail"
              >
                Form
              </label>
              <div
                className="col-sm-5"
              >
                <input
                  className="form-control"
                  id="formHorizontalEmail"
                  name="from"
                  onChange={[Function]}
                  placeholder="Date"
                  type="date"
                  value="2019-01-01"
                />
              </div>
              <label
                className="form-label col-form-label col-sm-1"
                htmlFor="formHorizontalEmail"
              >
                To
              </label>
              <div
                className="col-sm-5"
              >
                <input
                  className="form-control"
                  id="formHorizontalEmail"
                  name="to"
                  onChange={[Function]}
                  placeholder="Date"
                  type="date"
                  value="2000-01-01"
                />
              </div>
            </div>
          </form>
        </div>
        <div
          className="create-film col-md-4 col-12"
        >
          <a
            className="btn btn-primary"
            href="/add-movie"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Create
          </a>
        </div>
      </div>
      <div
        className="movies row"
      >
        <div
          className="col-md-4 col-sm-12 col-12"
        >
          <div
            className="card"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Die Another Day"
              className="card-img-top"
              onClick={[Function]}
              src="https://cdn.pocket-lint.com/r/s/660x/assets/images/148096-tv-feature-bond-gallery-image21-saq5axcp9j.jpg?v1"
            />
            <div
              className="card-body"
            >
              <div
                className="card-title h5"
              >
                Die Another Day
              </div>
              <p
                className="card-text"
              >
                20 November 2002
              </p>
              <p
                className="card-text"
              >
                Pierce Brosnan
              </p>
              <p
                className="card-text"
              />
            </div>
            <div
              className="card-footer"
              onClick={[Function]}
            >
              <i
                className="fa fa-heart"
                style={
                  Object {
                    "color": "black",
                  }
                }
              />
              <small
                className="text-muted"
              >
                 Press to add to favourite list
              </small>
            </div>
          </div>
        </div>
        <div
          className="col-md-4 col-sm-12 col-12"
        >
          <div
            className="card"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Casino Royale"
              className="card-img-top"
              onClick={[Function]}
              src="https://cdn.pocket-lint.com/r/s/660x/assets/images/148096-tv-feature-bond-gallery-image22-jnqmh3jklf.jpg?v1"
            />
            <div
              className="card-body"
            >
              <div
                className="card-title h5"
              >
                Casino Royale
              </div>
              <p
                className="card-text"
              >
                16 November 2006
              </p>
              <p
                className="card-text"
              >
                Daniel Craig
              </p>
              <p
                className="card-text"
              />
            </div>
            <div
              className="card-footer"
              onClick={[Function]}
            >
              <i
                className="fa fa-heart"
                style={
                  Object {
                    "color": "black",
                  }
                }
              />
              <small
                className="text-muted"
              >
                 Press to add to favourite list
              </small>
            </div>
          </div>
        </div>
        <div
          className="col-md-4 col-sm-12 col-12"
        >
          <div
            className="card"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Quantum of Solace"
              className="card-img-top"
              onClick={[Function]}
              src="https://cdn.pocket-lint.com/r/s/660x/assets/images/148096-tv-feature-bond-gallery-image23-074kml2fqy.jpg?v1"
            />
            <div
              className="card-body"
            >
              <div
                className="card-title h5"
              >
                Quantum of Solace
              </div>
              <p
                className="card-text"
              >
                31 October 2008
              </p>
              <p
                className="card-text"
              >
                Daniel Craig
              </p>
              <p
                className="card-text"
              />
            </div>
            <div
              className="card-footer"
              onClick={[Function]}
            >
              <i
                className="fa fa-heart"
                style={
                  Object {
                    "color": "black",
                  }
                }
              />
              <small
                className="text-muted"
              >
                 Press to add to favourite list
              </small>
            </div>
          </div>
        </div>
        <div
          className="col-md-4 col-sm-12 col-12"
        >
          <div
            className="card"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Skyfall"
              className="card-img-top"
              onClick={[Function]}
              src="https://cdn.pocket-lint.com/r/s/660x/assets/images/148096-tv-feature-bond-gallery-image24-avqr79slhy.jpg?v1"
            />
            <div
              className="card-body"
            >
              <div
                className="card-title h5"
              >
                Skyfall
              </div>
              <p
                className="card-text"
              >
                26 October 2012
              </p>
              <p
                className="card-text"
              >
                Daniel Craig
              </p>
              <p
                className="card-text"
              />
            </div>
            <div
              className="card-footer"
              onClick={[Function]}
            >
              <i
                className="fa fa-heart"
                style={
                  Object {
                    "color": "black",
                  }
                }
              />
              <small
                className="text-muted"
              >
                 Press to add to favourite list
              </small>
            </div>
          </div>
        </div>
        <div
          className="col-md-4 col-sm-12 col-12"
        >
          <div
            className="card"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Spectre"
              className="card-img-top"
              onClick={[Function]}
              src="https://cdn.pocket-lint.com/r/s/660x/assets/images/148096-tv-feature-bond-gallery-image25-qoue1ji6vz.jpg?v1"
            />
            <div
              className="card-body"
            >
              <div
                className="card-title h5"
              >
                Spectre
              </div>
              <p
                className="card-text"
              >
                26 October 2015
              </p>
              <p
                className="card-text"
              >
                Daniel Craig
              </p>
              <p
                className="card-text"
              />
            </div>
            <div
              className="card-footer"
              onClick={[Function]}
            >
              <i
                className="fa fa-heart"
                style={
                  Object {
                    "color": "black",
                  }
                }
              />
              <small
                className="text-muted"
              >
                 Press to add to favourite list
              </small>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
